### PR TITLE
Fix bug in preferences

### PR DESCRIPTION
**Pull request by bennetelli**

### CODE REVIEW

**Comment by bennetelli** [comment]
camel-rewrite-preferences

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/editorsettings/CamelEditorSettingsPage.java [hunk 1]
```diff
@@ -204,8 +101,25 @@ public void apply() throws ConfigurationException {
         getCamelPreferenceService().setShowCamelIconInGutter(camelIconInGutterCheckBox.isSelected());
         getCamelPreferenceService().setChosenCamelIcon(camelIconsComboBox.getSelectedItem().toString());
         getCamelPreferenceService().setCustomIconFilePath(customIconButton.getText());
-        getCamelPreferenceService().setIgnorePropertyList(ignorePropertyTable.getIgnoredProperties());
-        getCamelPreferenceService().setExcludePropertyFiles(excludePropertyFileTable.getExcludePropertyFiles());
+
+        //setModified(true);
```
Comment (fharms) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/editorsettings/CamelEditorSettingsPage.java]: No need to keep this

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/editorsettings/CamelEditorSettingsPage.java [hunk 2]
```diff
@@ -220,18 +134,24 @@ public void reset() {
         camelIconsComboBox.setSelectedItem(getCamelPreferenceService().getChosenCamelIcon());
         customIconButton.setText(getCamelPreferenceService().getCustomIconFilePath());
         customIconButton.setEnabled("Custom Icon".equals(camelIconsComboBox.getSelectedItem()));
-        ignorePropertyTable.reset();
-        excludePropertyFileTable.reset();
-    }
 
-    CamelPreferenceService getCamelPreferenceService() {
-        return ServiceManager.getService(CamelPreferenceService.class);
+        //setModified(false);
```
Comment (fharms) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/editorsettings/CamelEditorSettingsPage.java]: No need to keep this

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java [hunk 3]
```diff
@@ -16,295 +16,55 @@
...
-
-            assertTrue(indexOfApplicationConfigurable > 0);
-            assertEquals("<applicationConfigurable id=\"camel\" displayName=\"Apache Camel\" groupId=\"language\"", trimmedStrings.get(indexOfApplicationConfigurable));
-            String secondLineOfApplicationConfigurable = trimmedStrings.get(indexOfApplicationConfigurable + 1);
-            assertEquals("instance=\"org.apache.camel.idea.preference.CamelPreferencePage\"/>", secondLineOfApplicationConfigurable);
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-    }
+//    public void testPluginXmlShouldContainPreferencesPage() {
```
Comment (fharms) [inline camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java]: if this is not used anymore, please remove it

Comment (fharms) [reply camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java]: or add comment why it's comment out

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/properties/CamelIgnoreAndExcludePage.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+    private CamelPreferenceService getCamelPreferenceService() {
+        return ServiceManager.getService(CamelPreferenceService.class);
+    }
+
+    private JPanel createExcludePropertiesFilesTable() {
+        final JPanel mainPanel = new JPanel(new GridLayout(1, 1));
+        mainPanel.setPreferredSize(JBUI.size(300, 300));
+        mainPanel.setBorder(BorderFactory.createEmptyBorder(5, 5, 0, 0));
+        excludePropertyFilePanel = new AddEditRemovePanel<String>(new IgnoredUrlsModel(), getCamelPreferenceService().getExcludePropertyFiles(), "Exclude property file list") {
```
Comment (fharms) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/properties/CamelIgnoreAndExcludePage.java]: You could move this out to it's own internal class and let it extend AddEditRemovePanel and re-use it below also I guess

**Comment by bennetelli** [comment]
What do you mean with "has setup a exclude list etc," ? The exclude and ignore list entries will be stored in the CamelPreferencesService as it was before. I just replaced our custom JBTable and TableModel with AddEditRemovePanels.

Hope this answers your question? I also checked if the changes will be persisted and available after IDEA restart. That works fine 👍

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/properties/CamelIgnoreAndExcludePage.java (hunk 4 above)
Comment (fharms) [reply camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/properties/CamelIgnoreAndExcludePage.java]: @bennetelli did you see this comment?

Comment (bennetelli) [reply camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/properties/CamelIgnoreAndExcludePage.java]: @fharms fixed :)

**Comment by fharms** [comment]
@bennetelli Remember to Squash your commits when merge the pull-request  :-)

**Pull request merged**